src/goto-instrument/code_contracts.cpp

```diff
@@ -961,3 +950,723 @@ bool code_contractst::enforce_contracts(
+  : assigns_clause_targett(
+      Struct,
+      pointer_for(object_ptr),
+      contract,
+      log_parameter),
+    main_struct_standin(typet())
+{
+  const symbolt &struct_symbol =
+    contract.get_namespace().lookup(to_tag_type(object_ptr.type()));
+  const symbolt &function_symbol = contract.get_namespace().lookup(function_id);
+
+  // Declare a new symbol to stand in for the reference
+  symbolt struct_temp_symbol = contract.new_tmp_symbol(
+    pointer_object.type(),
+    function_symbol.location,
+    function_id,
+    function_symbol.mode);
+  main_struct_standin = struct_temp_symbol.symbol_expr();
+  local_standin_variables.push_back(main_struct_standin);
+
+  // Build standin variable initialization block
+  init_block.add(
+    goto_programt::make_decl(main_struct_standin, function_symbol.location));
+  init_block.add(goto_programt::make_assignment(
+    code_assignt(main_struct_standin, pointer_object),
+    function_symbol.location));
+
+  // Handle component members
+  std::vector<exprt> component_members;
+  const struct_typet &struct_type = to_struct_type(struct_symbol.type);
+  for(struct_union_typet::componentt component : struct_type.components())
+  {
+    exprt current_member = member_exprt(object_ptr, component);
+    component_members.push_back(current_member);
+  }
+
+  while(!component_members.empty())
+  {
+    exprt current_operation = component_members.front();
+    exprt operation_address = pointer_for(current_operation);
+
+    // Declare a new symbol to stand in for the reference
+    symbolt standin_symbol = contract.new_tmp_symbol(
+      operation_address.type(),
+      function_symbol.location,
+      function_id,
+      function_symbol.mode);
+
+    symbol_exprt current_standin = standin_symbol.symbol_expr();
+    local_standin_variables.push_back(current_standin);
+
+    // Add to standin variable initialization block
+    init_block.add(
+      goto_programt::make_decl(current_standin, function_symbol.location));
+    init_block.add(goto_programt::make_assignment(
+      code_assignt(current_standin, operation_address),
+      function_symbol.location));
+
+    if(current_operation.type().id() == ID_struct_tag)
+    {
+      const symbolt &current_struct_symbol =
+        contract.get_namespace().lookup(to_tag_type(current_operation.type()));
+
+      const struct_typet &curr_struct_t =
+        to_struct_type(current_struct_symbol.type);
+      for(struct_union_typet::componentt component : curr_struct_t.components())
+      {
+        exprt current_member = member_exprt(current_operation, component);
+        component_members.push_back(current_member);
+      }
+    }
+    component_members.erase(component_members.begin());
+  }
+}
+
+std::vector<symbol_exprt>
+assigns_clause_struct_targett::temporary_declarations() const
+{
+  return local_standin_variables;
+}
+
+exprt assigns_clause_struct_targett::alias_expression(const exprt &ptr)
+{
+  exprt running = false_exprt();
+  bool first_it = true;
```

Comment:
Replace these by `exprt::operandst disjuncts; disjuncts.reserve(local_standin_variables.size());` and ... (continues below)